### PR TITLE
OGL: Fall back to the old dual-source blending behaviour.

### PR DESCRIPTION
**Pull request by Armada651**
This allows us to still use dual-source blending in cases where it is required for accurate emulation and not just for small precision enhancements where it could crash some driver.

I also added OS X Intel to the bug, as those drivers seem to be affected by a similar bug. The OS X Nvidia drivers are not affected. I've tested it on my own MacBook Pro and confirmed this PR works correctly.

---

This change is [Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/4402)

### CODE REVIEW

File: Source/Core/VideoCommon/PixelShaderGen.cpp [hunk 1]
```diff
@@ -523,6 +525,7 @@ ShaderCode GeneratePixelShaderCode(APIType ApiType, const pixel_shader_uid_data*
     else
     {
       out.Write("FRAGMENT_OUTPUT_LOCATION(0) out vec4 ocol0;\n");
+      out.Write("vec4 ocol1;\n");  // Consume the output we don't use
```
Comment (Armada651) [inline Source/Core/VideoCommon/PixelShaderGen.cpp]: This is a bit of a hack, but it's so much nicer than checking these conditions every time or passing a boolean everywhere.

**Comment by ligfx** [comment]
I'd love to talk about this some more. I've heard a lot about how much better dual-source blending is, will turning it off on OSX lead to a performance hit?

If so, I'm still interested in the solution in #4397, which would let us keep dual-source turned on. We could probably come up with some numbers to better quantify "compiling a bunch of extra shaders:" in the FIFO log I've been testing, for example, only 1/28 shaders is used both with dual-source blending and normal blending.

**Comment by Armada651** [comment]
@ligfx This doesn't turn off dual-source blending, it just doesn't use it for minor blending precision anymore on driver that can't handle it correctly. Dual-source blending is still used in cases where the feature is required for emulation.

**Comment by ligfx** [comment]
@Armada651 So no performance hit?

**Comment by Armada651** [comment]
@ligfx No performance hit, just some minor color differences invisible to the naked eye.

**Comment by ligfx** [comment]
Okay great! Thanks for answering my questions.

**Pull request merged**